app/src/main/java/tech/relaycorp/gateway/pdc/local/routes/ParcelCollectionRoute.kt

```diff
@@ -28,53 +32,86 @@ class ParcelCollectionRoute
     private val collectParcelsProvider: Provider<CollectParcels>
 ) : PDCServerRoute {
 
+    @VisibleForTesting
+    var logger: Logger = Logger.getLogger(javaClass.name)
+
     private val asyncJob = SupervisorJob()
     private val asyncScope = CoroutineScope(asyncJob)
 
     override fun register(routing: Routing) {
-        routing.webSocket(URL_PATH) {
-            if (call.request.header(HEADER_ORIGIN) != null) {
-                // The client is most likely a (malicious) web page
-                close(
-                    CloseReason(
-                        CloseReason.Codes.VIOLATED_POLICY,
-                        "Web browser requests are disabled for security reasons"
-                    )
-                )
-                return@webSocket
+        try {
+            routing.webSocket(URL_PATH) {
+                try {
+                    handle()
+                } catch (exc: ClosedReceiveChannelException) {
+                    val reason = closeReason.await()
+                    if (reason == null) {
+                        logger.log(Level.WARNING, "TCP connection closed abruptly", exc)
+                    } else if (reason.knownReason != CloseReason.Codes.NORMAL) {
+                        logger.log(Level.WARNING, "TCP connection closed due to $reason", exc)
+                    }
+                } catch (exc: Exception) {
```

Comment:
I think it'd be best not to gracefully log and discard any exception, but limit that "graceful" handling to the exception(s) we really want to treat that way. We could be hiding bugs by doing this.

Reply:
This was my un-finished attempt to catch an abrupt client shutdown. Unsuccessful as well...